### PR TITLE
add option to listen to commit progress

### LINKED ISSUE #98: Question: Re
Hi Steve,

I am planning to put in a PR to address [this issue](https://github.com/commitizen/cz-cli/issues/8) but wanted to ask for some input beforehand. When using .commit(), it is possible that a repo will have pre-commit hooks configured. These hooks run before gulp-git's commit() callback is executed. 

These hooks can, and often are long running if, for example, they execute a bunch of tests. Currently, the output gets swallowed until the commit is finished, almost appearing as if the process has hung. Then all of a sudden all of the results appear on the screen in one large block. 

In these scenarios I think we'd ideally be able to incrementally pipe the output of the exec command to the console, or even better, emit a 'data' event that sends an incremental progress update.

I think I'm leaning towards the second option but I wanted to confirm that this was a good choice. The problem is, I'm not sure how to use self.emit to send the data back out of .commit(). I see self.emit('end') but obviously it doesn't pass anything out. Do you know if this is possible? 

Here is an example of what the change would ideally look like:

```
// Currently we don't do anything with the returned value of exec:
// exec(...) {}

// In the new changes we'd get the return value...
var execLogger = exec(...) {}

// Then listen to the data as it comes back...

// Something new came from stdout
execLogger.stdout.on('data',function(data){

  // Option 1: We could just log out the data as it comes back. This is intrusive, however.
  console.log(data);
});

// Something new came from stderr
execLogger.stderr.on('data',function(data){
  
  // Option 2: We could emit the data back up, allowing the consumer to ignore it or not 
  self.emit('data', data);
});

```

Anyhow I know that most of this works, I just wasn't sure of the gulp specifics surrounding .emit. Thanks.

### PR DESCRIPTION
**Pull request by jimthedev**
Closes #98
Adds tests, readme and code that allows for watching data events during git commit.

### CODE REVIEW

**Comment by jimthedev** [comment]
Ok, travis was having issues with 0.12.0 but it worked locally for me no problem. Should be all set now.

**Comment by stevelacy** [comment]
Will it not emit the data and return the large buffer at the end?

**Comment by jimthedev** [comment]
It could I suppose. Would it do it on .end?

File: lib/commit.js [hunk 1]
```diff
@@ -65,13 +65,25 @@ module.exports = function(message, opt) {
       cmd += '-a ' + opt.args + (opt.args.indexOf('--no-edit') === -1 ? ' --no-edit' : '');
     }
     var self = this;
-    exec(cmd, {cwd: opt.cwd, maxBuffer: opt.maxBuffer}, function(err, stdout, stderr){
+    var execChildProcess = exec(cmd, {cwd: opt.cwd, maxBuffer: opt.maxBuffer}, function(err, stdout, stderr){
       if (err) return cb(err);
       if (!opt.quiet) gutil.log(stdout, stderr);
```
Comment (stevelacy) [inline lib/commit.js]: This will log out the entire stdout as well as the emitData call

**Comment by stevelacy** [comment]
I'm referring to the gutil.log line which will dump the stdout as well as the incremental log if `quiet: true` is set

**Comment by jimthedev** [comment]
Hmm not sure I understand. Don't you mean if `quiet: false` then it will emit both? 

Even if it emits the event, it is up to the listener to console.log it or do whatever else. If they want to just ignore the data then that'd be ok.

**Comment by stevelacy** [comment]
published to 1.6.0

**Comment by jimthedev** [comment]
Thanks Steve!

**Pull request merged**